### PR TITLE
plugins: add description of the new JSON attribute

### PR DESCRIPTION
**Pull request by pryrt**
per https://github.com/notepad-plus-plus/nppPluginList/pull/423

@rdipardo, please review my changes, and let me know if it needs improvement.

### CODE REVIEW

**Comment by rdipardo** [comment]
@pryrt
Nice work.
I'm afraid, however, it's been obsoleted by notepad-plus-plus/notepad-plus-plus@a84cbe8f

### TL;DR
The `"npp-compatible-versions"` property will not be sufficient to avoid crashes.
Every plugin author must _also_ provide something called `"old-versions-compatibility"`, in the form of `[previous version(s)][previously compatible Npp versions]`, e.g. (for mimeTools), `"[,2.6][,8.2.1]"`

### Background 
There was a chance to make version checking _logical_ and _stable_, as I futilely tried to explain in past days:

- https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11358#issuecomment-1062692927
- https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11356#issuecomment-1061723215

Instead we now have a "system" that depends on plugin authors _telling_ N++ "this one is good, _that one_ is bad."

And when a plugin author forgets the other half, or gets it wrong? I guess we just tell users, "Oh, someone forgot/mistyped the 'old-versions-compatibility' thing"

- And what "someone" is that?
- Dunno; try removing plugins one-by-one to find out!

With "stability" like that, I'd say we're right back to where we started before notepad-plus-plus/nppPluginList#416.

**Comment by pryrt** [comment]
I think I've updated the PR to describe that second parameter as well.  Let me know if it still is incomplete or inaccurate, or whether the needs change again before the next Notepad++/nppPluginsList release.  (I will wait to merge this PR until the next release.)

**Comment by rdipardo** [comment]
- Looks good. (Can professional documentation turn sloppy, impromptu additions to the schema into a "specification"? Time will tell.) 

- I can only vouch for the patterns accepted by the "npp-compatible-versions" regex. I have not and will not be testing the "old-versions-compatibility"` thing. There are [online tools](https://regex101.com) for that.

- Requests for edits are pouring in from plugin authors at notepad-plus-plus/nppPluginList#424. At least wait for the final version to get merged. I predict there will be structural changes made after this hits end user PCs, so _well after_ the next release might be prudent

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
Thank you @pryrt !
